be/src/formats/csv/array_converter.cpp

```diff
@@ -67,28 +65,49 @@ bool ArrayConverter::read_quoted_string(Column* column, Slice s, const Options&
     return read_string(column, s, options);
 }
 
-bool ArrayConverter::_split_array_elements(Slice s, std::vector<Slice>* elements) const {
-    bool in_quote = false;
-    int array_nest_level = 0;
-    elements->push_back(s);
-    for (size_t i = 0; i < s.size; i++) {
-        char c = s[i];
-        // TODO(zhuming): handle escaped double quotes
-        if (c == '"') {
-            in_quote = !in_quote;
-        } else if (!in_quote && c == '[') {
-            array_nest_level++;
-        } else if (!in_quote && c == ']') {
-            array_nest_level--;
-        } else if (!in_quote && array_nest_level == 0 && c == ',') {
-            elements->back().remove_suffix(s.size - i);
-            elements->push_back(Slice(s.data + i + 1, s.size - i - 1));
-        }
+std::unique_ptr<ArrayReader> create_array_reader(const Converter::Options& options) {
+    std::unique_ptr<ArrayReader> array_reader;
+    if (options.array_format_type == ArrayFormatType::HIVE) {
+        char delimiter = get_collection_delimiter(options.array_hive_collection_delimiter,
+                                                  options.array_hive_mapkey_delimiter, options.array_hive_nested_level);
+        array_reader = std::make_unique<HiveTextArrayReader>(delimiter);
+    } else {
+        array_reader = std::make_unique<DefaultArrayReader>();
     }
-    if (array_nest_level != 0 || in_quote) {
-        return false;
+    return array_reader;
+}
+
+char get_collection_delimiter(char collection_delimiter, char mapkey_delimiter, size_t nested_array_level) {
```

Comment:
How about make it as a static method in `HiveArrayReader` because only used by it.